### DISCUSSION TITLE
Flutter for TV support

### DISCUSSION THREAD

**Discussion by MarioMisseyer** [codemagic-ci-cd/codemagic-docs#2089] [Ideas]
### Use case

```Text
Using Flutter for multiplatform development it is also possible to develop for Android TV and Flutter-Tizen with the same code base
With the current machine images you also build and publish Android TV apps but not a Flutter-Tizen build
```

### Idea

```Text
Have Flutter-Tizen also available on VM images to build a Tizen tpk artifact.
```

**Comment by Ycoi** [comment]
Thanks for the idea @MarioMisseyer! You should actually be able to do this by installing the needed tools to the VM as part of your workflow in custom scripts. If you bump into any issues with this you can always open a separate discussion in the Q&A section and our support will help you out.

If you set this up with Codemagic, would you be interested in writing an article about it for our [blog](https://blog.codemagic.io/)? If you’re interested, just respond me here and I can direct you to the correct people for more information. You can check existing blog posts for how a typical blog post looks like.

**Reply by MarioMisseyer** [reply]
I was more thinking to integrate it more and using:
` environment:`
`   flutter-tizen: stable`
`   tizen-sdk-tools:`
`     Gcc: 9.2`
`    dotnet:  7.0.401
`
Is it possible to get support to set this up ?

**Reply by Ycoi** [reply]
Sorry about the radio silence. Unfortuantely we are currently focusing on delivering other improvements and don’t currently have plans on adding this deep integration with flutter-tizen. The core strength of Codemagic is in building and publishing for mobile and that’s where our focus is on when building new features. 

Is there some special reason why installing the needed tools as part of the workflow wouldn’t work for you? If you encounter any issues with it you can always ask help in the Q&A section of this Github Discussions forum. 

You could for example set up a separate workflow for the tizen application while building your possible other flutter targets in a separarte workflow.

**Reply by MarioMisseyer** [reply]
We chose Codemagic because of Flutter support but we also chose Flutter because of multiplatform support including TV devices.
Another reason for choosing Codemagic is the lifecycle management of the test and build environment which saves us a lot of time.
So installing the tools ourselves will not give that benefit anymore.
We will start setting up a Tizen workflow with installing the tools but it would be much appreciated if we could collaborate on that.

**Reply by Ycoi** [reply]
Thanks for the clarification. At Codemagic we are big fans of Flutter and also Flutter multiplatform support and support building all the targets of default Flutter out of the box. Since Flutter for Tizen needs a separate tool installation, there are some extra considerations needed for possible support. Do you happen to know how often the flutter-tizen tools generally get updated and how the backwards compatibility usually is? 

I’ll leave this request open to see if there is more interest on Tizen support (this is the first time it has been requested). In the mean time indeed good if you try installing the tools on a separate workflow. And please share your experiences on how that goes.

**Reply by MarioMisseyer** [reply]
Looking at this Github I see 1-3 commits a month https://github.com/flutter-tizen/flutter-tizen

**Comment by mikhail-tokarev** [comment]
I will close this thread due to inactivity but let me highlight couple things:
1. we love Flutter 💙 we really do and it's always our priority
2. just because there are so many awesome Flutter tools for different devices we can't integrate with each of them
3. however it's possible to install and use everything you need from your yaml configuration.

While we cannot move forward with this particular proposal, we value our relationship and hope to explore other opportunities together in the future.
